specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2023-01-15-preview/failoverGroups.json

```diff
@@ -0,0 +1,231 @@
+        },
+        "spec": {
+          "$ref": "#/definitions/FailoverGroupSpec",
+          "description": "null",
+          "x-ms-client-flatten": false
+        },
+        "status": {
+          "$ref": "#/definitions/FailoverGroupStatus",
+          "description": "null",
+          "x-ms-client-flatten": false
+        }
+      },
+      "required": [
+        "partnerManagedInstanceId",
+        "spec"
+      ]
+    },
+    "FailoverGroupSpec": {
+      "description": "The specifications of the failover group resource.",
+      "type": "object",
+      "properties": {
+        "sharedName": {
+          "type": "string",
+          "description": "The shared name of the failover group for this SQL managed instance. Both SQL managed instance and its partner have to use the same shared name."
+        },
+        "sourceMI": {
+          "type": "string",
+          "description": "The name of the SQL managed instance with this failover group role."
+        },
+        "partnerMI": {
+          "type": "string",
+          "description": "The name of the partner SQL managed instance."
+        },
+        "partnerMirroringURL": {
+          "type": "string",
+          "description": "The mirroring endpoint URL of the partner SQL managed instance."
+        },
+        "partnerMirroringCert": {
+          "type": "string",
+          "description": "The mirroring endpoint public certificate for the partner SQL managed instance. Only PEM format is supported."
+        },
+        "partnerSyncMode": {
+          "type": "string",
+          "enum": [
+            "async",
+            "sync"
+          ],
+          "x-ms-enum": {
+            "name": "failoverGroupPartnerSyncMode",
+            "modelAsString": true
+          },
+          "default": "async",
+          "description": "The partner sync mode of the SQL managed instance."
+        },
+        "role": {
+          "type": "string",
+          "enum": [
+            "primary",
+            "secondary",
+            "force-primary-allow-data-loss",
+            "force-secondary"
+          ],
+          "x-ms-enum": {
+            "name": "instanceFailoverGroupRole",
+            "modelAsString": true
+          },
+          "default": "primary",
+          "description": "The role of the SQL managed instance in this failover group."
+        }
+      },
+      "additionalProperties": {},
+      "required": [
+        "role"
+      ]
+    },
+    "FailoverGroupStatus": {
+      "type": "object",
+      "description": "The status of the Kubernetes custom resource.",
+      "properties": {
+        "lastUpdateTime": {
+          "type": "string",
+          "description": "The time that the custom resource was last updated."
+        },
+        "observedGeneration": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The version of the replicaSet associated with the failover group custom resource."
+        },
+        "state": {
```

Comment:
>state [](http://example.com/codeflow?start=9&length=5)

Is this a candidate for an enum?

Reply:
No, we don't have a cap, limit, or set of values for the observed generation.

Reply:
Yes, I added the enum values, thanks